packages/adk/src/models/ai-adk.ts

```diff
@@ -0,0 +1,397 @@
+	}
+
+	/**
+	 * Returns empty array - following Python ADK pattern
+	 */
+	static override supportedModels(): string[] {
+		return [];
+	}
+
+	protected async *generateContentAsyncImpl(
+		request: LlmRequest,
+		stream = false,
+	): AsyncGenerator<LlmResponse, void, unknown> {
+		try {
+			// Transform ADK request to AI SDK format
+			const messages = this.transformMessages(request.contents || []);
+			const system = request.getSystemInstructionText();
+			const tools = this.transformTools(request.config?.tools);
+
+			if (stream) {
+				const result = streamText({
+					model: this.modelInstance,
+					messages,
+					system,
+					tools,
+				});
+
+				for await (const part of result.fullStream) {
+					const response = this.streamPartToLlmResponse(part);
+					this.ensureAdkCompatibility(response, request);
+					yield response;
+				}
+			} else {
+				const result = await generateText({
+					model: this.modelInstance,
+					messages,
+					system,
+					tools,
+				});
+
+				const response = this.resultToLlmResponse(result);
+				this.ensureAdkCompatibility(response, request);
+				yield response;
+			}
+		} catch (error) {
+			const errorResponse = new LlmResponse({
+				errorCode: "AI_SDK_ERROR",
+				errorMessage: `AI SDK Error: ${String(error)}`,
+				content: {
+					role: "model",
+					parts: [{ text: `Error: ${String(error)}` }],
+				},
+				finishReason: "STOP",
+			});
+			this.ensureAdkCompatibility(errorResponse, request);
+			yield errorResponse;
+		}
+	}
+
+	/**
+	 * Ensures response structure matches ADK's AutoFlow expectations
+	 */
+	private ensureAdkCompatibility(
+		response: LlmResponse,
+		request: LlmRequest,
+	): void {
+		// Ensure basic response structure
+		if (!response.content) {
+			response.content = { role: "model", parts: [] };
+		}
+		if (!response.content.role) {
+			response.content.role = "model";
+		}
+		if (!response.content.parts || !Array.isArray(response.content.parts)) {
+			response.content.parts = [];
+		}
+		if (!response.usageMetadata) {
+			response.usageMetadata = {
+				promptTokenCount: 0,
+				candidatesTokenCount: 0,
+				totalTokenCount: 0,
+			};
+		}
+		if (!response.finishReason) {
+			response.finishReason = "STOP";
+		}
+
+		// Set up toolsDict with callable tools
+		const toolsMap = new Map<string, BaseTool>();
+		if (request.toolsDict) {
+			for (const [toolName, tool] of Object.entries(request.toolsDict)) {
+				toolsMap.set(toolName, tool);
+			}
+		}
+		(response as any).toolsDict = toolsMap;
+
+		// Set up function calls array for compatibility
+		const functionCalls: any[] = [];
+		for (const part of response.content.parts) {
+			if ((part as any).functionCall) {
+				const fc = (part as any).functionCall;
+				functionCalls.push({
+					name: fc.name,
+					args: fc.args,
+					id: fc.id,
+				});
+			}
+		}
+		(response as any).functionCalls = functionCalls;
+
+		// Set up additional compatibility properties
+		(response as any).candidates = [
```

Comment:
there are anys everywhere in the code. those are smells of not proper typing / not strong quality in code